src/shipping/components/ShippingZoneAddWarehouseDialog/ShippingZoneAddWarehouseDialog.tsx

```diff
@@ -75,65 +64,59 @@ const ShippingZoneAddWarehouseDialog: React.FC<ShippingZoneAddWarehouseDialogPro
   const countryChoices = mapCountriesToChoices(countries);
 
   return (
-    <Dialog
-      classes={{ paper: classes.overflow }}
-      onClose={onClose}
-      open={open}
-      fullWidth
-      maxWidth="sm"
-    >
-      <DialogTitle disableTypography>
-        <FormattedMessage
-          id="yzYXW/"
-          defaultMessage="Create New Warehouse"
-          description="header, dialog"
-        />
-      </DialogTitle>
+    <DashboardModal onChange={onClose} open={open}>
       <Form initial={initialForm} onSubmit={handleSubmit}>
-        {({ change, data }) => {
+        {({ change, data, submit }) => {
           const handleCountrySelect = createSingleAutocompleteSelectHandler(
             change,
             setCountryDisplayName,
             countryChoices,
           );
 
           return (
-            <>
-              <DialogContent className={classes.overflow}>
-                <TextField
-                  fullWidth
-                  label={intl.formatMessage({
-                    id: "llBnr+",
-                    defaultMessage: "Warehouse Name",
-                  })}
-                  name="name"
-                  value={data.name}
-                  onChange={change}
-                />
-                <FormSpacer />
-                <Hr />
-                <FormSpacer />
-                <CompanyAddressForm
-                  countries={countryChoices}
-                  data={data}
-                  disabled={disabled}
-                  displayCountry={countryDisplayName}
-                  errors={errors}
-                  onChange={change}
-                  onCountryChange={handleCountrySelect}
+            <DashboardModal.Content __maxWidth={600} __width="calc(100% - 64px)">
```

Comment:
Suggestion: We use the same props for some of the dialogs, maybe we can make a default const somewhere and just import it?